sites/management/commands/scan.py

```diff
@@ -24,19 +32,54 @@ def pshtt(domain):
     return pshtt_results, stdout, stderr
 
 
-def is_onion_available(pshtt_results):
+def is_onion_loc_in_meta_tag(url: str) -> Optional[bool]:
+    """
+    Make request to target URL, parse page content and see if there is a
+    tag with format:
+
+    <meta http-equiv="onion-location" content="http://myonion.onion">
     """
-    For HTTPS sites, we inspect the headers to see if the
-    Onion-Location header is present, indicating that the
-    site is available as an onion service.
+    try:
+        r = requests.get(url, timeout=TIMEOUT_REQUESTS)
+        tree = html.fromstring(r.content)
+        tags = tree.xpath('//meta[@http-equiv="onion-location"]/@content')
+        if len(tags) >= 1:
+            return True
+    except (etree.ParserError, requests.exceptions.RequestException) as e:
+        # Error when requesting or parsing the page content, we log and
+        # continue on.
+        logger.error(e)
+        return None
+
+    return False
+
+
+def is_onion_available(pshtt_results) -> Optional[bool]:
+    """
+    For HTTPS sites, we see if an Onion-Location is provided, indicating that
+    the site is available as an onion service.
     """
     onion_available = False
 
+    # First we see if the header is provided.
     for key in ["https", "httpswww"]:
         try:
             headers = pshtt_results["endpoints"][key]["headers"]
             if 'onion-location' in set(k.lower() for k in headers):
                 onion_available = True
+                return onion_available
+        except KeyError:
+            pass
+
+    # If the header is not provided, it's possible the news organization
+    # has included it the HTML of the page in a meta tag using the `http-equiv`
+    # attribute.
+    for key in ["https", "httpswww"]:
```

Comment:
If we are still concerned about how much time the page-scanning operation takes, we might be able to reduce that time by only fetching one URL here. Pshtt has a system for determining the "canonical" URL for a domain (based on what redirects to what, and maybe some other factors) and if that one is https, then we could only scan that one. This potentially would save us from doing a fair number of scans, and would only be inaccurate if the onion location meta tag is absent from the site's canonical URL but _present_ in the same URL with/without the `www`.

Just an idea, though. It might not be really necessary.

Reply:
this is a good point and I thought worth implementing to reduce scan time, so I've refactored the logic in e8ec6bc0a696695ca5935fcc5aeff067aceb9da7 to use the canonical URL